src/libPMacc/include/memory/shared/Allocate.hpp

```diff
@@ -105,6 +105,27 @@ namespace shared
 
 /** allocate shared memory
  *
+ * Do not use this macro within a factory method. This macro contains a pre compiler
+ * counter which is only increased once per source code line.
+ * Please do also not assume that this macro can be used to create aliased pointer
```

Comment:
You can even write: `You must not use this macro to create an aliased pointer to shared memory.`